### PR TITLE
feat(ext/crypto): implement AES-GCM decryption

### PR DESCRIPTION
**Pull request by littledivy**
Towards #11690

### CODE REVIEW

**Review by SeanWykes — COMMENTED** [review]
nice.

**Comment by SeanWykes** [comment]
General question: Is there anywhere we can document the current restrictions, apart from the PRs which once merged become somewhat invisible? For example, AES-CTR also has restrictions on counter-size ..

File: ext/crypto/00_crypto.js [hunk 1]
```diff
@@ -630,6 +631,66 @@
...
+          if (normalizedAlgorithm.iv.byteLength !== 12) {
+            throw new DOMException(
+              "Initialization vector length not supported",
+              "NotSupportedError",
+            );
+          }
+
+          // 4.
+          if (normalizedAlgorithm.additionalData !== undefined) {
+            if (normalizedAlgorithm.additionalData.byteLength > (2 ** 64) - 1) {
```
[the remark this answers is not in the document]
Comment (bnoordhuis) [reply ext/crypto/00_crypto.js:670]: Not really, but I suspect it's copied over from the spec algorithm. It also mirrors a similar check in this file.

**Review by bnoordhuis — APPROVED** [review]
The PR LGTM (and thanks!) but I second the suggestion from @SeanWykes to de-duplicate the logic using generics.

File: ext/crypto/00_crypto.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (littledivy) [reply ext/crypto/00_crypto.js:670]: Yeah, it's just there for the sake of the spec algorithm...even `MAX_SAFE_INTEGER` is around ~8 petabytes of allocation :smile:

**Comment by littledivy** [comment]
> @littledivy - General question: Is there anywhere we can document the current restrictions on tag-size for AES-GCM, apart from the PRs which once merged become somewhat invisible?
> 
> AES-CTR has similar restrictions on counter-size ..

@SeanWykes I think these limitations can be in the manual once we add a WebCrypto section. (https://github.com/denoland/manual/issues/26)

**Pull request merged**